src/core/ext/filters/message_size/message_size_filter.cc

```diff
@@ -147,14 +149,31 @@ static void recv_message_ready(void* user_data, grpc_error* error) {
     GRPC_ERROR_REF(error);
   }
   // Invoke the next callback.
-  GRPC_CLOSURE_RUN(calld->next_recv_message_ready, error);
+  grpc_closure* closure = calld->next_recv_message_ready;
+  calld->next_recv_message_ready = nullptr;
+  if (calld->seen_recv_trailing_metadata) {
+    calld->seen_recv_trailing_metadata = false;
```

Comment:
Why is this needed?

If this is for the case where there may be another recv_message op after the current one, I don't think that can actually happen, because the application can't start a second recv_message op until this one returns.

Reply:
I thought the same thing but I was actually seeing the call combiner being started twice

Reply:
Debugged it. What's actually happening is this sequence of events -
1) We receive trailing metadata but delay it because we have a recv message OP.
2) We receive the message and schedule the trailing metadata closure on the call combiner.
3) We run the recv trailing metadata closure.
4) In the mean time, we have received another recv message op which again sees that recv trailing metadata has been received and schedules the trailing metadata closure again.

Reply:
Okay, that makes sense.

It's probably worth a comment here explaining why this is safe.  In particular, one might think that it's not actually safe to resume the recv_trailing_metadata_ready callback until after we know that *all* recv_message ops have completed.  But in practice, we can't do that, because we don't actually know whether or not there are going to be additional recv_message ops on the call.  However, what saves us here is that once the transport has invoked the recv_trailing_metadata_ready callback, any further recv_message ops will return with a null payload.  And since we know that we won't generate an error in recv_message_ready if the payload is null (because it can't possibly be over the configured size limit), we don't need to defer the recv_trailing_metadata_ready callback until we see that subsequent recv_message_ready callback.